test/Empty/Maybe.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE ScopedTypeVariables, TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Empty.Maybe
+( tests
+) where
+
+import Control.Carrier.Empty.Maybe
+import Control.Monad.Trans.Maybe
+import Pure
+import Test.QuickCheck.Poly
+import Test.Tasty
+import Test.Tasty.QuickCheck
+
+tests :: TestTree
+tests = testGroup "Empty.Maybe"
+  [ testProperty "empty annihilation" $
+    \ k -> empty_annihilation ((~=) @B) (applyFun @A k)
+  ]
+
+(~=) :: (Eq a, Show a) => EmptyC PureC a -> EmptyC PureC a -> Property
+m1 ~= m2 = run (runEmpty m1) === run (runEmpty m2)
+
+
+instance Arbitrary1 m => Arbitrary1 (EmptyC m) where
+  liftArbitrary genA = EmptyC . MaybeT <$> liftArbitrary @m (liftArbitrary @Maybe genA)
+  liftShrink shrinkA = map (EmptyC . MaybeT) . liftShrink (liftShrink shrinkA) . runEmpty
```

Comment:
This is the direct approach. Another approach that occurred to me would be building up programs using the constructors of the effects for the various carriers (in this case `empty` and `pure`) and binding/sequencing chains of them together. That would certainly work, but ultimately every such program ends up being represented as `m (Maybe a)`, so we can, and do, generate that directly and wrap it up instead.